### PR TITLE
GLOCKFILE: remove dependency on github.com/stretchr/testify

### PR DESCRIPTION
**Pull request by tamird**
Depends on https://github.com/robfig/glock/pull/38.

cc @RaduBerinde @jordanlewis 

Depending on `testify` because it is a test dependency of one of our dependencies is the tail wagging the dog, so I'd prefer we simply not depend on it.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9813)

### CODE REVIEW

File: GLOCKFILE [hunk 1]
```diff
@@ -84,35 +75,31 @@ github.com/mdempsky/unconvert beb68d938016d2dec1d1b078054f4d3db25f97be
...
-github.com/peterbourgon/g2s 5767a0b2078638d14800683fd0fe425604883f63
 github.com/petermattis/goid ba001f8780f3bf978180f390ad7b5bac39fbf70a
 github.com/pkg/errors 839d9e913e063e28dfd0e6c7b7512793e0a48be9
 github.com/prometheus/client_model fa8ad6fec33561be4280a8f0514318c79d7f6cb6
 github.com/prometheus/common 85637ea67b04b5c3bb25e671dacded2977f8f9f6
 github.com/rcrowley/go-metrics ab2277b1c5d15c3cba104e9cbddbdfc622df5ad8
-github.com/robfig/glock 7bc8ce51048e2adc11733f90a87b1c02fb7feebe
+github.com/robfig/glock a7c25f6eb431c4e526bcd30d0c4393f2cfe051b6
 github.com/rubyist/circuitbreaker 7e3e7fbe9c62b943d487af023566a79d9eb22d3b
+github.com/russross/blackfriday 35eb537633d9950afc8ae7bdf0edb6134584e9fc
```
Comment (tschottdorf) [inline GLOCKFILE]: Where's that from? Glock bump?

File: GLOCKFILE [hunk 2]
```diff
@@ -84,35 +75,31 @@ github.com/mdempsky/unconvert beb68d938016d2dec1d1b078054f4d3db25f97be
...
 github.com/sasha-s/go-deadlock 09aefc0ac06ad74d91ca31acdb11fc981c159149
 github.com/satori/go.uuid b061729afc07e77a8aa4fad0a2fd840958f1942a
+github.com/shurcooL/sanitized_anchor_name 1dba4b3954bc059efc3991ec364f9f9a35f597d2
```
Comment (tschottdorf) [inline GLOCKFILE]: Where's that from? Glock bump?

**Review by tbg — APPROVED** [review]

**Comment by tschottdorf** [comment]
Nice.

**Comment by RaduBerinde** [comment]
Oh, I thought we have to depend on it, but I guess we don't because it's only used in `basictracer-go` testing code.

Can we add `ExpectEqualInt`, `ExpectEqualStr` in `testutils` and use those? The "before" code was nicer..

**Comment by tschottdorf** [comment]
> Can we add ExpectEqualInt, ExpectEqualStr in testutils and use those? The "before" code was nicer..

Those are always going to be personal questions, but my preference is and will always be to avoid decorators for this trivial code because that little bit of nicer comes with a little bit of cognitive overhead that I never find worth it.

File: GLOCKFILE (hunk 1 above)
Comment (tamird) [inline GLOCKFILE]: This was a bug in my glock PR, which I just fixed.

File: GLOCKFILE (hunk 2 above)
Comment (tamird) [inline GLOCKFILE]: This was a bug in my glock PR, which I just fixed.

**Comment by tamird** [comment]
Also, let's not have that discussion here.

**Comment by RaduBerinde** [comment]
I find that stance mind-boggling, being forced to write the same kind of error message over and over is indefensible in my mind (and frankly encourages skimping on checks in tests). But whatever, we've been fighting too much for small stuff. LGTM

**Comment by tschottdorf** [comment]
LGTM but PR title is betraying the actual change here, which is making Glock better.

**Comment by tschottdorf** [comment]
Ditto commit message.

**Comment by RaduBerinde** [comment]
You made this use `tamird/glock` just so you can merge it faster? What's the rush?

**Comment by tschottdorf** [comment]
Second that, there's a chance that upstream won't merge your change due to compatibility and then we're awkwardly flip-flopping.

> I find that stance mind-boggling, being forced to write the same kind of error message over and over is indefensible in my mind.

Maybe you're right, and I just don't end up in those situations often enough.

**Comment by RaduBerinde** [comment]
> Second that, there's a chance that upstream won't merge your change due to compatibility and then we're awkwardly flip-flopping.


Oh, I see, it was to strongarm another PR https://github.com/cockroachdb/cockroach/pull/9763#issuecomment-252371739 haha. SMH

**Comment by tamird** [comment]
Yep, that's exactly right. That PR was the reason for doing this work in
the first place.

**Comment by tamird** [comment]
@RaduBerinde let me clarify: yes, this was motivated by the desire to prevent the increased use of testify in #9763.

As far as strong-arming goes, that’s exactly what I feel was happening in #9763: the argument there was that increasing the use of testify was OK because @jordanlewis had consulted with you, and that we already had testify in our GLOCKFILE. That reasoning is unfortunate in that it seems to willfully ignore the subtlety of the dependency situation (before this PR) where testify was being brought in as a dependency's test dependency.

Furthermore, the specific decision of introducing a new testing paradigm is something which should be discussed deliberately (and without entanglement to other work), because it potentially affects every contributor to the project. It also happens to be the case that this was discussed before and rejected, though I don't recall where that discussion was had.

Lastly, I want to stress that I'm not categorically against introducing and using testify, but I am against it happening on the basis of flawed logic, and that's what I merged this PR. Now, if someone would like to initiate that discussion, they can and should do so, but they'll need to do it explicitly (by adding it to the GLOCKFILE).

Hopefully that makes sense, and if not, let's discuss further privately.

**Comment by RaduBerinde** [comment]
Tamir,

I am sure Jordan would have agreed to change those calllsites anyway, especially given that Bram agreed with you, and given that you posted this PR. I would have definitely recommended that, if he was planning to consult with me again (I had no idea it was such a big deal when he first asked me).

What you accused him of, "sneaking in" stuff, was actually **exactly** what you did in this PR. You changed the glock dependency to your private fork *after* you got our LGTMs and merged it.  I don't think you would have gotten the OKs with the change in that form; definitely not mine.

And for what? Even in the "worst" case that Jordan would have merged his PR with a few more uses of these functions. What's the big deal? They're easy enough to change back. And they're not incorrect. It's a cosmetic, non-technical issue. Is this really what we should be spending our energy on?

**Comment by tamird** [comment]
Once more, the goal here was not to decide anything. The use of testify was previously discussed in https://github.com/cockroachdb/cockroach/pull/3945 and rejected. The fact that #9308 (which introduced the use of `assert`) was merged without discussion on the use of testify is evidence in support of the need for this GLOCKFILE change.

**Comment by vivekmenezes** [comment]
I do not like the way this has been handled. Very bad for team spirit!
There are ways of achieving the same goal. I urge everyone to get folks on
the same page without causing pain. There are people behind these changes,
and one ought to recognize that. Looking at this from a purely technical
perspective is narrow minded. Let's not repeat this again.

**Pull request merged**